### ISSUE TITLE
Style/IfUnlessModifier forces lines far past Layout/LineLength Max when the line matches an AllowedPatterns entry

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#15531]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

`Style/IfUnlessModifier`

### Steps to reproduce the problem

1. Create `.rubocop.yml`:

```yaml
Layout/LineLength:
  Max: 40
  AllowedPatterns:
    - 'MARKER'
```

2. Create `example.rb`:

```ruby
# frozen_string_literal: true

def check(cond)
  if cond
    do_something("MARKER_that_makes_this_line_long")
  end
end
```

3. Run `rubocop --only Style/IfUnlessModifier,Layout/LineLength example.rb`

### Expected behavior

No offense. The modifier form is 58 characters, well past `Max: 40`, so collapsing it makes the code longer than the configured limit. RuboCop 1.88.2 reports no offenses for this input.

### Actual behavior

```
C:  4:  3: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

Running with `-a` produces a 58-character line under `Max: 40`:

```ruby
# frozen_string_literal: true

def check(cond)
  do_something("MARKER_that_makes_this_line_long") if cond
end
```

This changed in 1.89.0 with #15493, which routed the modifier cops' line-fit check through `acceptable_line_length?`:

```ruby
def too_long_single_line?(node)
  range = node.source_range
  range.single_line? && !acceptable_line_length?(range.source_line, range.first_line)
end
```

`AllowedPatterns`, `AllowURI` and cop directives are exemptions meaning "do not report this existing line", which is not the same as "any length is acceptable here". Treating them as the latter means the modifier cops now actively rewrite code to exceed `Max`. The same reasoning applies to `Style/WhileUntilModifier` and `Style/GuardClause`, which #15493 changed as well.

This is not only a contrived case. Homebrew hit it on a real config where an `AllowedPatterns` entry is an unanchored regex, so a pattern intended for one construct matched a much wider set of lines, and the cop demanded modifier forms running to 179 characters against a `Max` of 118.

Would it make sense for the modifier cops to keep checking the joined line against `Max`, or failing that to make the new behavior configurable?

### RuboCop configuration

```yaml
Layout/LineLength:
  Max: 40
  AllowedPatterns:
    - 'MARKER'
```

### RuboCop version

```console
$ rubocop -V
1.89.0 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 4.0.6) [arm64-darwin20]
```
